test/GreetingBook.t.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.23;
+
+import {Test} from "forge-std/Test.sol";
+
+import {MockPortal} from "omni/core/test/utils/MockPortal.sol";
+
+import {GreetingBook} from "src/GreetingBook.sol";
+
+contract GreetingBookTest is Test {
+    address user = address(0x123);
+    uint64 gasLimit = 120_000;
+    uint64 mockSourceChainId = 1;
+
+    MockPortal portal;
+    GreetingBook greetingBook;
+
+    function setUp() public {
+        portal = new MockPortal();
+        greetingBook = new GreetingBook(address(portal));
+    }
+
+    function testXGreet() public {
+        string memory greeting = "Hello, world!";
+        portal.mockXCall(
+            mockSourceChainId,
+            address(portal),
+            address(greetingBook),
+            abi.encodeWithSelector(GreetingBook.greet.selector, user, greeting),
+            gasLimit
+        );
+        GreetingBook.Greeting memory lastGreet;
+        (lastGreet.user, lastGreet.message, lastGreet.sourceChainId, lastGreet.timestamp) = greetingBook.lastGreet();
+        assertEq(lastGreet.message, greeting);
```

Comment:
Thoughts on `portal.mockXCall` util?

Reply:
I love it

Reply:
Only thing that's a little confusing is the sender field. We've set it to the portal here, but that doesn't really make sense, in fact is probably an anti-pattern